com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/graph/ResourceGraph.java

```diff
@@ -0,0 +1,294 @@
+     * @return The resource node for the path
+     */
+    public ResourceNode getResourceNodeFromPath(String path) {
+        return pathToNodeLookup.get(path);
+    }
+
+    /**
+     * Set hex digests for all resource nodes in the graph
+     * @param hexDigests Map with hex digests, keyed on relative resource paths
+     */
+    public void setHexDigests(Map<String, String> hexDigests) {
+        for (ResourceNode resourceNode : resourceNodes) {
+            String hexDigest = hexDigests.get(resourceNode.getPath());
+            resourceNode.setHexDigest(hexDigest);
+        }
+    }
+
+    private void flagAllNonExcludedResources(ResourceNode node, HashSet<ResourceNode> visited) {
+        for (ResourceNode child : node.getChildren()) {
+            if (visited.contains(child)) {
+                continue;
+            }
+            child.flagAsUsedInMainBundle();
+            visited.add(child);
+            if (child.checkType(ResourceNode.Type.ExcludedCollectionProxy)) {
+                continue;
+            }
+            flagAllNonExcludedResources(child, visited);
+        }
+    }
+
+    /**
+     * Flag all resource nodes used in the main resource bundle.
+     * Resource nodes without this flag can be excluded from the main archive 
+     * and moved to a live update archive
+     */
+    public void findAllResourcesReferencedFromMainCollection() {
+        flagAllNonExcludedResources(root, new HashSet<ResourceNode>());
+    }
+
+    /**
+     * Create a list of all excluded resources. A resource is considered to be
+     * excluded if it is only referenced from excluded collection proxies. If a
+     * resource is referenced both from an excluded collection proxy and from
+     * a non-excluded collection it will not be considered an excluded resource.
+     * @return List of excluded resources
+     */
+    public List<String> createExcludedResourcesList() {
+        findAllResourcesReferencedFromMainCollection();
+        List<String> excludedResources = new LinkedList<>();
+        for (ResourceNode node : resourceNodes) {
+            if (node.isInMainBundle()) {
+                continue;
+            }
+            excludedResources.add(node.getPath());
+        }
+        return excludedResources;
+    }
+
+    public void writeJSON(ResourceNode node, JsonGenerator generator, boolean shouldPublishLU) throws IOException {
+        generator.writeStartObject();
+        generator.writeFieldName("path");
+        generator.writeString(node.getPath());
+
+        generator.writeFieldName("hexDigest");
+        generator.writeString(node.getHexDigest());
+
+        if (!node.checkType(ResourceNode.Type.None)) {
+            generator.writeFieldName("nodeType");
+            generator.writeString(node.getNodeTypeAsString());
+        }
+
+        if (shouldPublishLU) {
+            generator.writeFieldName("isInMainBundle");
+            generator.writeBoolean(node.isInMainBundle());
+        }
```

Comment:
@britzl `isInMainBundle` isn't useful if user doesn't use LiveUpdate and it maybe misleads for a developer. So, I don't include this data if LU is off.